src/UnderlineNav2/LoadingCounter.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import {useTheme, Theme} from '../ThemeProvider'
+
+export const LoadingCounter = () => {
+  const {theme} = useTheme()
+
+  const getLoaderStyle = (currentTheme?: Theme) => {
+    return `@keyframes loading {
+      from {
+        opacity: 0.4;
+      }
+      to {
+        opacity: 0.8;
+      }
+    }
+    animation: loading 1.2s linear infinite alternate;
+    background-color: ${currentTheme?.colors.neutral.emphasis};
+    border-color: ${currentTheme?.colors.border.default};
+    width: 22px;
+    height: 16px;
+    display: inline-block;
+    border-radius: 20px;`
+  }
+
+  const StyledLoadingCounter = styled.div`
```

Comment:
We want to avoid creating new components inside render functions because it will recreate this component on every render.

We could simplify this component like this:


```jsx
import {get} from '../constants'
import {keyframes} from 'styled-components'

const loading = keyframes`
  from { opacity: 0.4; }
  to { opacity: 0.8; }
`

export const LoadingCounter = styled.div`
  animation: ${loading} 1.2s linear infinite alternate;
  background-color: ${get('colors.neutral.emphasis')};
  border-color: ${get('colors.border.default')};
  ...
`
```

Reply:
Ahh thanks! This is a great feedback 🙌🏼